File: CryptoProject/Vignere.js
function vigenereEncrypt(plaintext, key) {
    let ciphertext = '';
    const keyLength = key.length;
    for (let i = 0; i < plaintext.length; i++) {
        const plainChar = plaintext[i];
        const keyChar = key[i % keyLength];
        let cipherChar;

        if (/[a-zA-Z]/.test(plainChar)) {
            const shift = keyChar.toLowerCase().charCodeAt(0) - 'a'.charCodeAt(0);
            const isUpperCase = plainChar === plainChar.toUpperCase();
            const baseCharCode = isUpperCase ? 'A'.charCodeAt(0) : 'a'.charCodeAt(0);
            cipherChar = String.fromCharCode(baseCharCode + (plainChar.charCodeAt(0) - baseCharCode + shift) % 26);
        } else {
            cipherChar = plainChar;
        }

        ciphertext += cipherChar;
    }
    return ciphertext;
}

document.getElementById("cipherForm3").addEventListener("submit", function (event) {
    event.preventDefault();

    const plaintext = document.getElementById("pt3").value;
    const keyword = parseInt(document.getElementById("k3").value);  
    let extendedKey = '';
        for (let i = 0; i < plaintext.length; i++) {
            extendedKey += keyword[i % keyword.length];
        }
        //console.log('Plaintext: ', plaintext);
        const ciphertext = vigenereEncrypt(plaintext, extendedKey);
        //console.log('Ciphertext: ', ciphertext);
        document.getElementById("outputDiv3").innerHTML = "Cipher text: " + ciphertext;
        rl.close();
});

File: CryptoProject/Vignere.cpp
#include <iostream>
#include <string>
using namespace std;

string vigenereEncrypt(string plaintext, string key) {
    string ciphertext;
    int keyLength = key.length();
    for (int i = 0; i < plaintext.length(); i++) {
        char plainChar = plaintext[i];
        char keyChar = key[i % keyLength];
        char cipherChar;

        if (isalpha(plainChar)) {
            int shift = keyChar - 'a';
            if (isupper(plainChar)) {
                cipherChar = 'A' + (plainChar - 'A' + shift) % 26;
            } else {
                cipherChar = 'a' + (plainChar - 'a' + shift) % 26;
            }
        } else {
            cipherChar = plainChar;
        }

        ciphertext += cipherChar;
    }
    return ciphertext;
}

int main() {
    string keyword, plaintext;
    cout << "Plaintext: ";
    cin >> plaintext;
    cout << "Key: ";
    cin >> keyword;
    string extendedKey;
    for (int i = 0; i < plaintext.length(); i++)
        extendedKey += keyword[i % keyword.length()];
    cout << "Plaintext: " << plaintext << endl;
    string ciphertext = vigenereEncrypt(plaintext, extendedKey);
    cout << "Ciphertext: " << ciphertext << endl;
    return 0;
}
